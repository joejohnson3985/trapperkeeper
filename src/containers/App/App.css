.create-new-icon:hover {
  fill: #f5bda8;
}
