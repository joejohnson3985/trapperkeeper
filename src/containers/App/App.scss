@import '../../Utilities/_Variables.scss';

body {
  background-color: $plain-background
}

::placeholder {
  color: $accent-color;
}