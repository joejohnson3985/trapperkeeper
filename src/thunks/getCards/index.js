import { setCards } from '../../actions';

const getCards = (url) => {
  return async (dispatch) =>  {
    try {
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const cards = await response.json();
      dispatch(setCards(cards))
    } catch (error) {
    }
  }
}

export default getCards;