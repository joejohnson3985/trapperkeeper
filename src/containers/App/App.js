import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import getCards from '../../thunks/getCards';
import Header from '../../components/Header';
import Form from '../Form';
import CardContainer from '../CardContainer';
import NotFound from '../../components/NotFound';
import './App.scss';

export class App extends Component {

  componentDidMount = () => {
    this.props.getCards('http://localhost:3000/api/v1/cards');
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Route exact path='/new-note' component={Form} />
        <Route exact path='/notes/:id' 
          render={({ match }) => {
            const card = this.props.cards.find(card => card.id === parseInt(match.params.id));
            return card ? <Form cardData={card} git/> : <NotFound />;
          }} 
        />
        <Switch>
          <Route path='/' exact component={CardContainer} />
          <Route exact path='/notes/:id' component={CardContainer} />
          <Route exact path='/new-note' component={CardContainer} />
          <Route path='' component={NotFound} />
        </Switch>
      </div>
    ); 
  }
}

App.propTypes = {
  cards: PropTypes.array,
  getCards: PropTypes.func
};

export const mapStateToProps = (state) => ({
  cards: state.cards
});

export const mapDispatchToProps = (dispatch) => ({
  getCards: (url) => dispatch(getCards(url))
});

export default connect(mapStateToProps, mapDispatchToProps)(App)
