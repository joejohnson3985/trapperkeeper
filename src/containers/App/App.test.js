import React from 'react';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { shallow } from 'enzyme';
import  getCards  from '../../thunks/getCards/index';
jest.mock('../../thunks/getCards/index');

describe('App', () => {

  describe('component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <App />
      )
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  });
      
  describe('mapStateToProps', () => {
    it('should have a mapped state prop', () => {
      let mockState ={
        cards: [],
        magic: []
      }
      let expected = {
        cards: []
      }
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  });

  describe('mapDispatchToProps', () => {
    it('should call getCards with the correct params', () => {
      const mockUrl = 'www.bob.com'
      const mockDispatch = jest.fn();
      const actionToDispatch = getCards(mockUrl)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.getCards(actionToDispatch)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  });
  
});
