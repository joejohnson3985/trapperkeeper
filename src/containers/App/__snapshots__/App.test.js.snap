// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match the snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Form)",
        "type": [Function],
      }
    }
    exact={true}
    path="/new-note"
  />
  <Route
    exact={true}
    path="/notes/:id"
    render={[Function]}
  />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CardContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CardContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/notes/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CardContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/new-note"
    />
    <Route
      component={[Function]}
      path=""
    />
  </Switch>
</div>
`;
